sched/sched/sched_releasetcb.c

```diff
@@ -154,7 +154,11 @@ int nxsched_release_tcb(FAR struct tcb_s *tcb, uint8_t ttype)
 #ifdef CONFIG_ARCH_ADDRENV
       /* Release this thread's reference to the address environment */
 
-      ret = up_addrenv_detach(tcb->group, tcb);
+      ret = up_addrenv_detach(tcb);
+      if (ttype == TCB_FLAG_TTYPE_TASK)
```

Comment:
how to handle the main thread exit before other threads in the same task

Reply:
It is fixed by the reference counter in the second patch. I think the 1st patch on itself does not work properly, the reference counter is needed as well. If you think it is better to merge / squash the first 2 patches I can do that ?

Reply:
Both patch is very big, let's keep as it.